### PR TITLE
Create index

### PR DESCRIPTION
**Pull request by Jesspinoza**

### CODE REVIEW

File: client/src/components/SaladListComponent.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from "react"
+import SaladTile from "./SaladTile"
+
+
+const SaladListComponent = (props) => {
```
Comment (dovakeen118) [inline client/src/components/SaladListComponent.js:4]: should only have one empty line separating - can remove either line 3 or 4
https://github.com/airbnb/javascript#whitespace--no-multiple-blanks

File: client/src/components/SaladListComponent.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    const [salads, setSalads] = useState([])
+
+    const getSalads = async() =>{
```
Comment (dovakeen118) [inline client/src/components/SaladListComponent.js]: should have a space before opening braces - `=> {`
https://github.com/airbnb/javascript#whitespace--before-blocks

should add a space before parenthesis - `async ()`
https://github.com/airbnb/javascript#whitespace--around-keywords

File: client/src/components/SaladListComponent.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        try{
+            const response = await fetch("/api/v1/salads")
+            if(!response.ok){
+                const errorMessage = `${response.status} (${response.statusText})`
+                const error = new Error(errorMessage)
+                throw(error)
+            }
+            const responseBody = await response.json()
+            setSalads(responseBody.salads)
+        }catch(error){
```
Comment (dovakeen118) [inline client/src/components/SaladListComponent.js]: spacing as well with try/catch braces

File: client/src/components/SaladListComponent.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        }
+    }
+
+    useEffect(()=> {
+        getSalads()
+    }, [])
+
+    const saladItems = salads.map((salad)=> {
+        
+        return (
```
Comment (dovakeen118) [inline client/src/components/SaladListComponent.js:28]: can remove the empty line before the return
https://github.com/airbnb/javascript#whitespace--padded-blocks

File: client/src/components/SaladListComponent.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            <SaladTile
+            key={salad.id}
+            salad={salad}
```
Comment (dovakeen118) [inline client/src/components/SaladListComponent.js]: props should be indented within the component
```js
<SaladTile
  key={salad.id}
  salad={salad}
/>
```

File: client/src/components/SaladTile.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { Link } from "react-router-dom"
+
+const SaladTile = (props) => {
+    const { name, description, id } = props.salad 
+    return (
+        <div>
+        <li>{name}</li>
+        <p>{description}</p>
```
Comment (dovakeen118) [inline client/src/components/SaladTile.js]: `li` and `p` should be indented as they are children of the `div`

File: client/src/components/SaladListComponent.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+            />
+        )
+    })
+
+    return (
+        <div>
+        <h1>My Salad List</h1>
+        <ul>{saladItems}</ul>
```
Comment (dovakeen118) [inline client/src/components/SaladListComponent.js]: the `h1` and `ul` should be indented as they are children of the `div`

consider changing the `ul` to a `div`, as an `ul` expects `li` elements as its children, but the `saladItems` are a collection of `div` elements

File: server/src/db/Seeder.js [hunk 8]
```diff
@@ -1,9 +1,23 @@
...
+    const sally = await User.query().insert({ email: "sallyseashell@email.com", cryptedPassword: "54321" })
+    const mary = await User.query().insert({ email: "Mary301@email.com", cryptedPassword: "24689" })
+
+    const cobb = await Salad.query().insert({ name: "Cobb", userId: john.id })
+    const fruit = await Salad.query().insert({ name: "Fruit", userId: john.id })
+    const potato = await Salad.query().insert({ name: "Potato", userId: mary.id })
+    const quinoa = await Salad.query().insert({ name: "Quinoa", userId: mary.id })
+    const greek = await Salad.query().insert({ name: "Greek", userId: mary.id })
+    const caesar = await Salad.query().insert({ name: "Caesar", userId: sally.id })
+    const kale = await Salad.query().insert({ name: "Kale", userId: sally.id })
```
Comment (dovakeen118) [inline server/src/db/Seeder.js]: we should have separate seeders for each entity - UserSeeder, SaladSeeder, etc. Ideally should be set up to ensure duplicate records are not created if the seeder is run multiple times

File: server/src/models/Salad.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
+const Model = require("./Model.js")
+
+
+class Salad extends Model{
```
Comment (dovakeen118) [inline server/src/models/Salad.js:3]: only one empty line for separation (lines 2 & 3)

File: server/src/models/Salad.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        }
+    }
+
+    static get jsonSchema() {
+        return {
+            type: 'object',
+            required: ["name"],
+            properties: {
+                name: {type: "string"},
+                description: {type: "string"}
```
Comment (dovakeen118) [inline server/src/models/Salad.js]: info in braces should be space padded - `{ type: "string" }`
https://github.com/airbnb/javascript#whitespace--in-braces

File: server/src/routes/api/v1/saladsRouter.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+
+import { Salad } from "../../../models/index.js"
+
+const saladsRouter = new express.Router()
+
+saladsRouter.get("/", async (req, res) => {
+    try{
+        const salads = await Salad.query()
+        return res.status(200).json({ salads: salads })
+    }catch (error){
```
Comment (dovakeen118) [inline server/src/routes/api/v1/saladsRouter.js]: spacing around try/catch braces
```js
try {
  ...
} catch (error) {
  ...
}
```

**Review by dovakeen118 — COMMENTED** [review]
work wise this looks great and is scoped well for the feature! the comments are super nit picking best practices following the JS style guide

File: client/src/components/SaladListComponent.js [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+                salad={salad}
+            />
+        )
+    })
+
+    return (
+        <div>
+            <h1>My Salad List</h1>
+            <ul>{saladItems}</ul>
+        </div>
```
Comment (nwalberts) [inline client/src/components/SaladListComponent.js]: this div isnt doing anything if it doesn have attributes. Delete it and replace with a React fragment e.g. `<></>`

File: client/src/components/SaladListComponent.js [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+    )
+
```
Comment (nwalberts) [inline client/src/components/SaladListComponent.js]: delete this extra line

File: client/src/components/SaladListComponent.js [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+    return (
+        <div>
+            <h1>My Salad List</h1>
+            <ul>{saladItems}</ul>
+        </div>
+    )
+    
+}
+
+
```
Comment (nwalberts) [inline client/src/components/SaladListComponent.js]: delete one of these two empty lines

File: client/src/components/SaladTile.js [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+        </div>
```
Comment (nwalberts) [inline client/src/components/SaladTile.js:10]: this component looks like it still needs the indentation that Kerrin requested

Additionally, the div you have isnt doing anything, so please replace with a fragment

File: server/src/models/Salad.js [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+            type: 'object',
+            required: ["name"],
+            properties: {
+                name: { type: "string" },
+                description: { type: "string" }
+            }
+
+        }
+
+    }
```
Comment (nwalberts) [inline server/src/models/Salad.js:32]: remove empty lines 31 and 33

File: server/src/routes/api/v1/saladsRouter.js [hunk 17]
```diff
@@ -0,0 +1,18 @@
...
+saladsRouter.get("/", async (req, res) => {
+    try {
+        const salads = await Salad.query()
+        return res.status(200).json({ salads: salads })
+    } catch (error) {
+        return res.status(500).json({ errors: error })
+    }
+})
+
+
```
Comment (nwalberts) [inline server/src/routes/api/v1/saladsRouter.js:17]: remove one empty line

File: server/src/routes/rootRouter.js [hunk 18]
```diff
@@ -1,12 +1,15 @@
...
 import userSessionsRouter from "./api/v1/userSessionsRouter.js";
 import usersRouter from "./api/v1/usersRouter.js";
 import clientRouter from "./clientRouter.js";
+
 const rootRouter = new express.Router();
 rootRouter.use("/", clientRouter);
 rootRouter.use("/api/v1/user-sessions", userSessionsRouter);
 rootRouter.use("/api/v1/users", usersRouter);
+rootRouter.use("/api/v1/salads", saladsRouter);
+
```
Comment (nwalberts) [inline server/src/routes/rootRouter.js:12]: this empty line isnt needed, remove

**Review by nwalberts — CHANGES_REQUESTED** [review]

**Review by meehanti23 — COMMENTED** [review]
looks good

**Review by bobbydigital1986 — COMMENTED** [review]
looks good

**Pull request merged**